source/grid/tria.cc

```diff
@@ -3317,8 +3320,11 @@ namespace internal
                                                           line[2]->index(),
                                                           line[3]->index(),
                                                           "boundary ids"));
-
-                quad->set_boundary_id_internal(subcell_quad.boundary_id);
+                // do not use the boundary id provided in subcell_quad
+                // if it is an invalid or internal one
+                if (subcell_quad.boundary_id !=
+                    numbers::internal_face_boundary_id)
+                  quad->set_boundary_id_internal(subcell_quad.boundary_id);
```

Comment:
Let's use `TriaAccessor::set_boundary_id` then.
```suggestion
                  quad->set_boundary_id(subcell_quad.boundary_id);
```